daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -283,7 +284,31 @@ class Compiler private (var validateDFDLSchemas: Boolean,
 
   def reload(savedParser: java.nio.channels.ReadableByteChannel): DFDL.DataProcessor = {
     try {
-      val objInput = new ObjectInputStream(new GZIPInputStream(Channels.newInputStream(savedParser))) {
+      val is = Channels.newInputStream(savedParser)
+
+      // Read the magic number and version information for this saved parser
+      // directly from the input stream. That information is not compressed or
+      // serialized
+      val magicNumber = "DAFFODIL "
```

Comment:
calling this "magic number" is just confusing. It's not a number, and it's not like magic numbers that are just a specific 2 bytes at start of a file. 

Call it requiredDataPrefixString or something clearer.